historycombobox.cpp

```diff
@@ -0,0 +1,67 @@
+/**
+ * \file
+ * <!--
+ * Copyright 2015 Develer S.r.l. (http://www.develer.com/)
+ * -->
+ *
+ * \brief HistoryComboBox class implementation
+ *
+ * \author Aurelien Rainone <aurelien@develer.org>
+ */
+
+#include "historycombobox.h"
+#include <history.h>
+#include <algorithm>
+#include <QLineEdit>
+
+
+HistoryComboBox::HistoryComboBox(QWidget *parent) :
+    QComboBox(parent)
+{
+    history = new History(this);
+}
+
+void HistoryComboBox::fillList(QString current_text)
+{
+    QStringList hist_lines(history->getHistory());
+    std::reverse(hist_lines.begin(), hist_lines.end());
```

Comment:
Si hai totalemente raggione, possiamo salvare il ```std::reverse``` call, fatto ad ogni modifica della lista, se gestiamo la lista in ordine reverse dall inizio.